src/validator.ts

```diff
@@ -1,7 +1,9 @@
+import { VerbType } from './authorization';
+
 export interface Validator {
     /**
      * returns a resolved Promise if the resource is valid. Otherwise throws an error
      * @throws InvalidResourceError
      */
-    validate(resource: any, tenantId?: string): Promise<void>;
+    validate(resource: any, params: { tenantId?: string; httpVerb?: VerbType }): Promise<void>;
```

Comment:
Let's use `'create'|'update'` instead of httpVerb. It's better if our core logic is separate from the HTTP layer

https://github.com/awslabs/fhir-works-on-aws-interface/blob/fdb4ee588328b1ab04e2f178e9c18c0b4f8a354c/src/constants.ts#L55-L56

Also, It's better to make `params` itself optional since all it's keys are optional